source/config.js

```diff
@@ -281,3 +282,5 @@ if (fs.existsSync(path.join(__dirname, '..', '.git'))){
 } else {
   module.exports.ungitDevVersion = module.exports.ungitPackageVersion;
 }
+
+module.exports.isGitOptionalLocks = semver.satisfies(module.exports.gitVersion, '2.15.0');
```

Comment:
IMHO you could simply set `GIT_OPTIONAL_LOCKS=0` on the parent's env and be done with it. I don't think that `git` will look twice if it doesn't know what  `GIT_OPTIONAL_LOCKS` is

Reply:
My original thought is to limit this to `git status` only as that is the biggest offender for this issue.  

If we do that it will be applying to all git operations right?  I maybe too cautious but I would like to limit this to an end point and observe

Reply:
Hmm so are you suggesting that we set env `GIT_OPTIONAL_LOCKS=0` env variable and override this env variable settings by doing `--optional-locks` on `rebase`, `merge`, and `cherry-pick`?  (I would also add `patch` to list as well)

Reply:
I think I didn't explain myself thoroughly.

If you think the `git` repository as a FSM, all those operations (`rebase`, `merge`, `cherry-pick`, `patch`, `am` ... etc) "change the repository's state" before they return back to the "idle" state when they are finished.

If you run on Unix, and you have git_ps1 set, you will see it.
(Examples [this](https://pagefaultblog.files.wordpress.com/2017/02/git-prompt2.png?w=685) from [here](https://pagefault.blog/2017/02/16/improved-git-cli-with-git-completion-and-git-prompt/), or [that](https://raw.githubusercontent.com/arialdomartini/oh-my-git-gh-pages/master/images/samples/14-action-in-progress.jpg) from [there](https://github.com/arialdomartini/oh-my-git))

_During_ that time, _it would be nice_ to set `GIT_OPTIONAL_LOCKS=0` on the `git`'s environment.

----

In that sense, you can have the best of two worlds: _Not changing_ current behavior and opportunistically still updating index, _unless_ in "a critical state" in the repository.

Reply:
Considering the doc explicitly calls out git status running on background, let's do this first without setting env for other operations.

https://git-scm.com/docs/git-status#_background_refresh